tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Proxy/InferredProxyData.cs

```diff
@@ -0,0 +1,38 @@
+// <copyright file="InferredProxyData.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Proxy;
+
+/// <summary>
+/// Represents the inferred proxy data that is used to create an inferred span that represents the proxy.
+/// </summary>
+/// <param name="proxyName">The name/type of the proxy service (e.g. "aws-apigateway")</param>
+/// <param name="startTime">When the proxy received the request</param>
+/// <param name="domainName">Domain name of the proxy service</param>
+/// <param name="httpMethod">HTTP method of the original request</param>
+/// <param name="path">Request path at the proxy</param>
+/// <param name="stage">Deployment stage name</param>
+internal class InferredProxyData(string proxyName, DateTimeOffset startTime, string domainName, string httpMethod, string path, string stage)
+{
+    // x-dd-proxy
+    public string ProxyName { get; init; } = proxyName;
+
+    // x-dd-proxy-request-time-ms
+    public DateTimeOffset StartTime { get; init; } = startTime;
+
+    // x-dd-proxy-domain-name
+    public string DomainName { get; init; } = domainName;
+
+    // x-dd-proxy-httpmethod
+    public string HttpMethod { get; init; } = httpMethod;
+
+    // x-dd-proxy-path
+    public string Path { get; init; } = path;
+
+    // x-dd-proxy-stage
+    public string Stage { get; init; } = stage;
```

Comment:
I assume this is all already defined (and implemented?) in an RFC, but that's a lot of extra headers, I'm surprised if they can't be at least partially combined?

Reply:
It wasn't clear to me from the dev docs how to achieve this: https://docs.aws.amazon.com/apigateway/latest/developerguide/request-response-data-mappings.html . I did look into it by testing API Gateway directly but wasn't able to prove that it works.

For now, I think we should stick with these separate headers and in the future, if someone has time to research the exact configuration needed to collapse the headers into a single value, we can refactor/improve this with another header configuration. It is possible other gateway systems handle headers differently.

Another pro of having them split out for now is that there's no confusion to what the header needs to be.

Reply:
> that's a lot of extra headers

I commented the same thing in the RFC, but several languages were already implementing the feature at that point 😅 

> For now, I think we should stick with these separate headers and in the future...

Ideally, we should figure these things out before releasing this feature as GA, because once we have tracer versions deployed in the wild, even if we change the headers, we'll need to support these ones for a long time to avoid breaking changes. Is this less of an issue in AWS Lambda since users just need to update the Lambda layer?

Reply:
I may be mistaken but I think it has already been released.